packages/dds/matrix/src/matrix.ts

```diff
@@ -475,6 +482,9 @@ export class SharedMatrix<T = any>
 		if (count === 0) {
 			return;
 		}
+		if (rowStart > this.rowCount) {
+			throw new UsageError("insertRows: out of bounds");
```

Comment:
I like the idea of adding more information - assuming it gets propagated. `insertRows: out of bounds rowStart ${rowStart}, rowCount ${this.rowCount}`